### PR TITLE
CMR-9850: Update usage of Testcontainers

### PR DESCRIPTION
**Pull request by TylerHeald1**
# Overview
Updating usage of Testcontainers
### What is the feature/fix?

updating our usage of testcontainers a bit

### What is the Solution?

Changed what we look for for container startup, but also the ordering of withStartupTimeout. Set startup timeout gets overwritten by setting a waitingFor WaitStrategy.
Also added a docker compose file that can potentially be used for local dev and addressed some linter output

### What areas of the application does this impact?

Integration tests, local dev

# Checklist

- [ ] I have updated/added unit and integration tests that prove my fix is effective or that my feature works
- [ ] New and existing unit and int tests pass locally and remotely with my changes
- [ ] I have removed unnecessary/dead code and imports in files I have changed
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

File: dev-system/src/cmr/dev_system/system.clj [hunk 1]
```diff
@@ -117,7 +117,7 @@
 
 (defmethod create-elastic :external
   [_]
-  (elastic-config/set-elastic-port! 9209))
```
Comment (jmaeng72) [inline dev-system/src/cmr/dev_system/system.clj:120]: so we are no longer supporting external elastic? If so, then should we just get rid of this method entirely?

File: dev-system/compose.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+services:
```
Comment (jmaeng72) [inline dev-system/compose.yaml:1]: When you say this docker compose can be 'potentially' used for local dev. Are you going to add how to do this in README?

File: dev-system/src/cmr/dev_system/system.clj (hunk 1 above)
Comment (TylerHeald1) [reply dev-system/src/cmr/dev_system/system.clj:120]: We can use external elastic just fine, it just doesn't seem there's a real need to set the port to 9209 when you can just use 9210 which is the default that CMR checks for elasticsearch on

File: dev-system/compose.yaml (hunk 2 above)
Comment (TylerHeald1) [reply dev-system/compose.yaml:1]: Sure I can add a bit of documentation

**Review by jmaeng72 — APPROVED** [review]

**Review by eereiter — APPROVED** [review]

**Pull request merged**